src/misc/ErrorHandlerImpl.ts

```diff
@@ -72,16 +74,37 @@ export async function handleUncaughtErrorImpl(e: Error) {
 	} else if (e instanceof SessionExpiredError) {
 		reloginForExpiredSession()
 	} else if (e instanceof OutOfSyncError) {
-
-		// When the user logs in and their offline database is out of sync, we just silently purge it and continue
-		// If the user is not using an offline database and they just go out of sync due to being logged in for too long,
-		// then they should log out and in again
-		const isUsingOfflineStorage = isOfflineStorageAvailable() && logins.isUserLoggedIn() && logins.getUserController().sessionType === SessionType.Persistent
-		if (isUsingOfflineStorage) {
-			Dialog.message("dataExpiredOfflineDb_msg")
-		} else {
-			Dialog.message("dataExpired_msg")
-		}
+		const isOffline = isOfflineStorageAvailable() && logins.isUserLoggedIn() && logins.getUserController().sessionType === SessionType.Persistent
+		Dialog.showActionDialog({
```

Comment:
Is there a reason to not use `message` dialog? It looks a bit weird with Primary button.

Reply:
I found the corner button to be too unimposing, also having the button at the bottom kinda forces the user to at least skim the text rather than just ignore it (maybe? im not a UX psychologist). To me this dialog says "we're gonna log you out and delete your offline data, no objections". I mentioned this in the PR message if you read it

Reply:
worth noting that we have a similar dialog for 2FA, maybe they could be consolidated (if we keep it like this)